nix-rust/src/util/base32.rs

```diff
@@ -64,7 +65,7 @@ pub fn decode(input: &str) -> Result<Vec<u8>, crate::Error> {
     let mut nr_bits_left: usize = 0;
     let mut bits_left: u16 = 0;
 
-    for c in input.chars().rev() {
+    for c in input.chars() {
```

Comment:
Hm, is this correct? IIRC we really do need to process it in reverse order (our base32 encoding is pretty quirky).